### PR TITLE
Refactor snarked ledger timeout in tests

### PR DESCRIPTION
**Pull request by deepthiskumar**
Calculate snarked ledger timeouts using test config and slot fill rate
Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #0000

### CODE REVIEW

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Comment by deepthiskumar** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!ci-build-me

**Review by nholland94 — APPROVED** [review]

**Comment by deepthiskumar** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
this also needs to go into compatible as well.

File: src/lib/integration_test_lib/wait_condition.ml [hunk 1]
```diff
@@ -202,11 +201,19 @@ struct
...
+      Test_config.slots_for_blocks (num_proofs - 1)
+    in
+    let total_slots = slots_for_first_proof + slots_for_additional_proofs in
     { id = Ledger_proofs_emitted_since_genesis
     ; description
     ; predicate = Network_state_predicate (check (), check)
-    ; soft_timeout
-    ; hard_timeout
+    ; soft_timeout = Network_time_span.Slots (total_slots + 6)
+    ; hard_timeout = Network_time_span.Slots (total_slots + 12)
```
Comment (QuiteStochastic) [inline src/lib/integration_test_lib/wait_condition.ml:216]: why is it `total_slots+6` and `total_slots+12`?  shouldn't it be based on a multiplier, like `(total_slots * 1.5)` and `(total_slots *2)`?

**Review by QuiteStochastic — APPROVED** [review]

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Comment by deepthiskumar** [comment]
!ci-build-me

**Pull request merged**